src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -2,6 +2,25 @@
 
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        if (names.length == 0 || data.length == 0 || dateFrom.isEmpty() || dateTo.isEmpty()) {
+            return null;
+        }
+        StringBuilder results = new StringBuilder("Report for period ");
+        int firstIndex = dateFrom.indexOf('=') + 3;
+        results.append(dateFrom.substring(firstIndex, dateFrom.lastIndexOf(dateFrom)))
+                .append(" - ")
+                .append(dateTo.substring(firstIndex, dateTo.lastIndexOf(dateTo)))
+                .append(System.lineSeparator());
+        for (int i = 0; i < data.length; i++) {
```

Comment:
This loop seems to be redundant as you are not using the 'i' index to access the 'data' array properly. You should iterate over 'names' and for each name, calculate the salary based on the 'data' array.